ttg/ttg/parsec/ttg.h

```diff
@@ -3502,6 +3503,78 @@ namespace ttg_parsec {
 
 #include "ttg/make_tt.h"
 
+  namespace device {
+    namespace detail {
+      static std::size_t nb_ttg_devices = 0;
+      static int *parsec_devid;
+      static int *ttg_devid;
+    }
+
+    std::size_t nb() {
+      if(0 != detail::nb_ttg_devices) return detail::nb_ttg_devices;
+      detail::ttg_devid = new int[parsec_nb_devices];
+      for(int i = 0; i < parsec_nb_devices; i++) {
+        parsec_device_module_t *m = parsec_mca_device_get(i);
+        if(m->type == PARSEC_DEV_CPU || m->type == PARSEC_DEV_CUDA) { 
+          detail::ttg_devid[i] = detail::nb_ttg_devices;
+          detail::nb_ttg_devices++; 
+          continue; 
+        }
+        detail::ttg_devid[i] = -1;
+      }
+      detail::parsec_devid = new int[detail::nb_ttg_devices];
```

Comment:
Mhh, this means we always have to call `nb()` to get a valid pointer, before we can call `allocate()`.

Reply:
Yes, I'm aware of that, that's why `allocate()` and `free()` (and other functions that rely on the conversion) always do check `if( nb() >= ... )` as the first instruction before using `detail::parsec_devid` ... But maybe it's too circumvoluted... 

There are alternatives... Let me know if you want to discuss this.